markovify/text.py

```diff
@@ -9,6 +11,18 @@
 DEFAULT_MAX_OVERLAP_TOTAL = 15
 DEFAULT_TRIES = 10
 
+try:
+    # From relevant section of Python docs:
+    # "These functions control how a process is allocated CPU time by the
+    # operating system. They are only available on some Unix platforms.
+    # For more detailed information, consult your Unix manpages."
+    DEFAULT_MP_POOL_SIZE = len(os.sched_getaffinity(0))
+except AttributeError:
+    DEFAULT_MP_POOL_SIZE = 4
+
+# TODO: find something which cannot be present in input
+END = "KTHXBYE"
```

Comment:
`END` is a sentinel we need to inject in queues in order to let workers exit from `iter` like [here](https://github.com/jsvine/markovify/pull/173/files#diff-cfe54a37d7e936c77ed4911fccde1a2440243885ac8d7f8427d9dcff7baca7d7R205). Since queues get feeded with sentences eg. [here](https://github.com/jsvine/markovify/pull/173/files#diff-cfe54a37d7e936c77ed4911fccde1a2440243885ac8d7f8427d9dcff7baca7d7R249), the ideal sentinel would be something we know we cannot get during sentences split. I've marked a TODO for an help: do you think an "invalid" multibyte string could do instead of this placeholder? Or anything else is sensible not to expect as an input sentence. Should it not be feasible, ie. we can't find something which fits, another implementation should be researched, perhaps using [`JoinableQueue`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.JoinableQueue)

Reply:
`markovify` currently uses these markers: https://github.com/jsvine/markovify/blob/2d1fb783ceb6d46baf891fedf72bf13d02b55f43/markovify/chain.py#L7-L8

Is that the sort of thing you're aiming to do here, or is it slightly different?